lib/Demangling/Demangler.cpp

```diff
@@ -2509,9 +2509,47 @@ NodePointer Demangler::demangleThunkOrSpecialization() {
         return nullptr;
       result = addChild(result, kind);
       result = addChild(result, demangleIndexSubset());
-      if (!nextIf('p')) return nullptr;
+      if (!nextIf('p'))
+        return nullptr;
+      result = addChild(result, demangleIndexSubset());
+      if (!nextIf('r'))
+        return nullptr;
+      return result;
+    }
+    case 'U': {
+      auto result = createNode(
+          Node::Kind::AutoDiffSelfReorderingReabstractionThunk);
+      auto optionalGenSig = popNode(Node::Kind::DependentGenericSignature);
+      auto toType = popNode(Node::Kind::Type);
+      auto fromType = popNode(Node::Kind::Type);
+      result = addChild(result, fromType);
+      result = addChild(result, toType);
+      addChild(result, optionalGenSig);
+      auto kind = demangleAutoDiffFunctionKind();
+      if (!kind)
```

Comment:
That check is not needed, because if kind is null, `addChild` will return null

Reply:
Fixed.